### PR TITLE
Work towards PHPStan level 7

### PR DESCRIPTION
**Pull request by leofeyer**

### CODE REVIEW

File: core-bundle/tests/Controller/ContentElement/TableControllerTest.php [hunk 1]
```diff
@@ -104,7 +107,10 @@ public function testOutputsTableWithHeaderAndFooter(): void
             </div>
             HTML;
 
-        $this->assertSameHtml($expectedOutput, $response->getContent());
+        $html = $response->getContent();
+
+        $this->assertNotFalse($html);
+        $this->assertSameHtml($expectedOutput, $html);
```
Comment (ausi) [inline core-bundle/tests/Controller/ContentElement/TableControllerTest.php]: I think I’d change the `assertSameHtml()` method here instead:

```php
protected function assertSameHtml(string $expected, mixed $actual, string $message = ''): void
{
    $this->assertIsString($actual);
    $this->assertSame(
        $this->normalizeWhiteSpaces($expected),
        $this->normalizeWhiteSpaces($actual),
        $message
    );
}
```
WDYT?

Comment (leofeyer) [reply core-bundle/tests/Controller/ContentElement/TableControllerTest.php]: Changed in 8b52d39a40b3958570399bf26f47a1fc868f72b4.

**Comment by leofeyer** [comment]
Frankly, I don‘t know if I have the time and willpower to ever finish this PR. There are so many tedious things to fix such as this one:



The return type of Symfony‘s [`Response::getContent()`](https://github.com/symfony/symfony/blob/ccd6a30665c7473876bfb34e1945c5e5fe713099/src/Symfony/Component/HttpFoundation/Response.php#L452-L455) method is set to `string|false` even though `Response::$content` is a string and the setter only allows `?string $content`. So the return type is simply wrong but to fix it, I would have to make a PR for Symfony, which is additional work and might not be merged until Symfony 7, because it is an API change.

How can we handle such cases?

**Comment by ausi** [comment]
> The return type of Symfony‘s [`Response::getContent()`](https://github.com/symfony/symfony/blob/ccd6a30665c7473876bfb34e1945c5e5fe713099/src/Symfony/Component/HttpFoundation/Response.php#L452-L455) method is set to `string|false` even though…

I think this is correct because `Response` is a parent class, see for example the `BinaryFileResponse` child class: https://github.com/symfony/symfony/blob/ccd6a30665c7473876bfb34e1945c5e5fe713099/src/Symfony/Component/HttpFoundation/BinaryFileResponse.php#L354-L357

> How can we handle such cases?

If an external library would have wrong return types we would need to provide a stub file that overwrites the method with the correct type information. But we should not do this often I think as the maintenance of the stub files will create additional work.

**Comment by leofeyer** [comment]
> I think this is correct because `Response` is a parent class, see for example the `BinaryFileResponse` child class

Oh, I missed that. Thank you for the heads-up.

But this implies that we have to handle the `false` case if we want to unlock level 7, even though we know that the method never actually returns `false`. That seems like pointless extra work to me. 🤷‍♂️

**Comment by ausi** [comment]
> That seems like pointless extra work to me. 🤷‍♂️

Agree. Especially for the test methods we often don’t need this extra level of type safety IMO. Can we disable this check for the tests so that we can get the src to level 7?

**Comment by leofeyer** [comment]
Nope, I‘m giving up. That is not worth the effort. 🤷‍♂️

**Pull request closed without merging**